File: ServerSpace/src/main/java/DataLayer/TarifaDB.java
package DataLayer;

import EntityLayer.EntidadEntity;
import EntityLayer.TarifaEntity;
import Enumerados.ProcessActionEnum;
import Framework.BaseDB;
import Framework.Inj;
import Framework.Utilidades;
import Framework.injector;

import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;

public class TarifaDB extends BaseDB {

    injector InjDB = new injector();

    public ArrayList<TarifaEntity> ObtenerMain() {

        ArrayList<TarifaEntity> DatoMemoria = new ArrayList<>();
        TarifaEntity en;
        try {
            InjDB.Sp("sp_TarifaObtenerMain");
            ResultSet rs = InjDB.RunSelect();
            fillSchemeTable(rs);
            while (rs.next()) {
                en = new TarifaEntity();
                if (fillFrom(rs, en)) {
                    DatoMemoria.add(en);
                }
            }

        } catch (SQLException e) {
            System.out.println("ERROR " + e);
            throw new UnsupportedOperationException("Datalater :" + e);
        }
        return DatoMemoria;
    }

    public TarifaEntity Registrar(TarifaEntity entity) {
        Boolean State = null;
        try {
            Inj.IniciarTranssaccion(false);
            String Store = "sp_TarifaRegistrar";
            if (entity.getAction() == ProcessActionEnum.Update.getValor()) {
                Store = "sp_TarifaActualizar";
            }
            Inj.Sp(Store);
            Inj.Pmt_Integer("@TarifaId", entity.getTarifaId(), true);
            Inj.Pmt_Integer("@ObjetoRefereciaId", entity.getObjetoReferenciaId(), false);
            Inj.Pmt_String("@Correlativo", entity.getCorrelativo(), false);
            Inj.Pmt_Integer("@MonedaId", entity.getMonedaId(), false);
            Inj.Pmt_Integer("@TipoTarifaEnum", entity.getTipoTarifaEnum(), false);
            Inj.Pmt_Integer("@UnidadMedidaId", entity.getUnidadMedidaId(), false);
            Inj.Pmt_Double("@CostoTarifa", entity.getCostoTarifa(), false);
            Inj.Pmt_String("@NomTarifa", entity.getNomTarifa(), false);
            Inj.Pmt_String("@NomComercial", entity.getNomComercial(), false);
            Inj.Pmt_String("@Descripcion", entity.getDescripcion(), false);
            Inj.Pmt_String("@FechaRegistro", Utilidades.getFechaRegistro(), false);
            Inj.Pmt_String("@CodUsuario", entity.getCodUsuario(), false);
            Inj.Pmt_Boolean("@EstadoRegistro", entity.getEstadoRegistro(), false);
            if (entity.getAction() == ProcessActionEnum.Add.getValor()) {
                int Id = Inj.RunInsert();
                State = Id > 0;
                if (State) {
                    entity.setTarifaId(Id);
                }
            }
            if (entity.getAction() == ProcessActionEnum.Update.getValor()) {
                Inj.RunUpdate();
            }

        } catch (Exception ex) {
            throw new UnsupportedOperationException("Datalater : " + ex);
        }
        return entity;
    }

    public ArrayList<TarifaEntity> ObtenerItem(int TarifaId) {

        ArrayList<TarifaEntity> DatoMemoria = new ArrayList<>();
        TarifaEntity en;
        try {
            InjDB.Sp("sp_TarifaObtenerItem");
            InjDB.Pmt_Integer("@TarifaId", TarifaId, false);
            ResultSet rs = InjDB.RunSelect();
            fillSchemeTable(rs);
            while (rs.next()) {
                en = new TarifaEntity();
                if (fillFrom(rs, en)) {
                    DatoMemoria.add(en);
                }
            }

        } catch (SQLException e) {
            System.out.println("ERROR " + e);
            throw new UnsupportedOperationException("Datalater :" + e);
        }
        return DatoMemoria;
    }

    public ArrayList<TarifaEntity> BuscarRecurso(String Nombre, int Tipo) {

        ArrayList<TarifaEntity> DatoMemoria = new ArrayList<>();
        TarifaEntity en;
        try {
            InjDB.Sp("sp_TarifaBuscarRecurso");
            InjDB.Pmt_String("@Nombre", Nombre, false);
            InjDB.Pmt_Integer("@Tipo", Tipo, false);
            ResultSet rs = InjDB.RunSelect();
            fillSchemeTable(rs);
            while (rs.next()) {
                en = new TarifaEntity();
                if (fillFrom(rs, en)) {
                    DatoMemoria.add(en);
                }
            }

        } catch (SQLException e) {
            System.out.println("ERROR " + e);
            throw new UnsupportedOperationException("Datalater :" + e);
        }
        return DatoMemoria;
    }
    
    public ArrayList<TarifaEntity> GetTarifaBuscarItem(String NomComercial) {
        injector InjDB = new injector();
        ArrayList<TarifaEntity> DatoMemoria = new ArrayList<>();
        TarifaEntity en;
        try {
            InjDB.Sp("sp_TarifaBuscarItem");
            InjDB.Pmt_String("@NomComercial", NomComercial , false);
            ResultSet rs = InjDB.RunSelect();
            fillSchemeTable(rs);
            while (rs.next()) {
                en = new TarifaEntity();
                if (fillFrom(rs, en)) {
                    DatoMemoria.add(en);
                }
            }
        } catch (SQLException e) {
            System.out.println("ERROR " + e);
            throw new UnsupportedOperationException("Datalater :" + e);
        }
        return DatoMemoria;
    }
}

File: Space/Space.DataLayer/TarifaDB.cs
using Framework;
using Space.EntityLayer;
using System;
using System.Collections.Generic;
using System.Data;
using System.Data.SqlClient;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Space.DataLayer
{
    public class TarifaDB : BaseDB
    {
        public List<TarifaEntity> ObtenerMain()
        {
            try
            {

                List<TarifaEntity> ListaData = new List<TarifaEntity>();
                Conexion.OpenConnection();
                Conexion.Store("sp_TarifaObtenerMain");
                SqlDataReader reader = Conexion.ReaderData();
                FillSchemeTable(reader);
                while (reader.Read())
                {
                    TarifaEntity entity = new TarifaEntity();
                    if (FillFrom<TarifaEntity>(reader, entity)) ListaData.Add(entity);
                }
                reader.Close();
                Conexion.CloseConnection();
                return ListaData;
            }
            catch (Exception ex)
            {
                Conexion.Restore();
                throw ex;
            }
        }

        public Int32 Registrar(TarifaEntity Ent)
        {
            try
            {

                String NameStore = "sp_TarifaRegistrar";

                Conexion.Store(NameStore);
                Conexion.AddParemeter("TarifaId", SqlDbType.Int, 10, ParameterDirection.InputOutput, Ent.TarifaId);
                Conexion.AddParemeter("ObjetoRefereciaId", SqlDbType.Int, 10, ParameterDirection.Input, Ent.ObjetoReferenciaId);
                Conexion.AddParemeter("Correlativo", SqlDbType.VarChar, 200, ParameterDirection.Input, Ent.Correlativo);
                Conexion.AddParemeter("MonedaId", SqlDbType.Int, 10, ParameterDirection.Input, Ent.MonedaId);
                Conexion.AddParemeter("TipoTarifaEnum", SqlDbType.Int, 10, ParameterDirection.Input, Ent.TipoTarifaEnum);
                Conexion.AddParemeter("UnidadMedidaId", SqlDbType.Int, 10, ParameterDirection.Input, Ent.UnidadMedidaId);
                Conexion.AddParemeter("CostoTarifa", SqlDbType.Decimal, 17, ParameterDirection.Input, Ent.CostoTarifa);
                Conexion.AddParemeter("NomTarifa", SqlDbType.VarChar, 200, ParameterDirection.Input, Ent.NomTarifa);
                Conexion.AddParemeter("NomComercial", SqlDbType.VarChar, 200, ParameterDirection.Input, Ent.NomComercial);
                Conexion.AddParemeter("Descripcion", SqlDbType.VarChar, 200, ParameterDirection.Input, Ent.Descripcion);
                Conexion.AddParemeter("FechaRegistro", SqlDbType.DateTime, 15, ParameterDirection.Input, Ent.FechaRegistro);
                Conexion.AddParemeter("CodUsuario", SqlDbType.VarChar, 20, ParameterDirection.Input, Ent.CodUsuario);
                Conexion.AddParemeter("EstadoRegistro", SqlDbType.VarChar, 20, ParameterDirection.Input, Ent.EstadoRegistro);
                Ent.TarifaId = Conexion.ExecuteQueryItem("TarifaId");

                return Ent.TarifaId;
            }
            catch (Exception ex)
            {
                Conexion.Restore();
                throw ex;
            }
        }

        public Int32 Actualizar(TarifaEntity Ent)
        {
            try
            {

                String NameStore = "sp_TarifaActualizar";

                Conexion.Store(NameStore);
                Conexion.AddParemeter("TarifaId", SqlDbType.Int, 10, ParameterDirection.InputOutput, Ent.TarifaId);
                Conexion.AddParemeter("ObjetoRefereciaId", SqlDbType.Int, 10, ParameterDirection.Input, Ent.ObjetoReferenciaId);
                Conexion.AddParemeter("Correlativo", SqlDbType.VarChar, 200, ParameterDirection.Input, Ent.Correlativo);
                Conexion.AddParemeter("MonedaId", SqlDbType.Int, 10, ParameterDirection.Input, Ent.MonedaId);
                Conexion.AddParemeter("TipoTarifaEnum", SqlDbType.Int, 10, ParameterDirection.Input, Ent.TipoTarifaEnum);
                Conexion.AddParemeter("UnidadMedidaId", SqlDbType.Int, 10, ParameterDirection.Input, Ent.UnidadMedidaId);
                Conexion.AddParemeter("CostoTarifa", SqlDbType.Int, 10, ParameterDirection.Input, Ent.CostoTarifa);
                Conexion.AddParemeter("NomTarifa", SqlDbType.VarChar, 200, ParameterDirection.Input, Ent.NomTarifa);
                Conexion.AddParemeter("NomComercial", SqlDbType.VarChar, 200, ParameterDirection.Input, Ent.NomComercial);
                Conexion.AddParemeter("Descripcion", SqlDbType.VarChar, 200, ParameterDirection.Input, Ent.Descripcion);
                Conexion.AddParemeter("FechaRegistro", SqlDbType.DateTime, 15, ParameterDirection.Input, Ent.FechaRegistro);
                Conexion.AddParemeter("CodUsuario", SqlDbType.VarChar, 20, ParameterDirection.Input, Ent.CodUsuario);
                Conexion.AddParemeter("EstadoRegistro", SqlDbType.VarChar, 20, ParameterDirection.Input, Ent.EstadoRegistro);
                Ent.TarifaId = Conexion.ExecuteQueryItem("TarifaId");

                return Ent.TarifaId;
            }
            catch (Exception ex)
            {
                Conexion.Restore();
                throw ex;
            }

        }

        public List<TarifaEntity> ObtenerItem(Int32 TarifaId)
        {
            try
            {

                List<TarifaEntity> ListaData = new List<TarifaEntity>();
                Conexion.OpenConnection();
                Conexion.Store("sp_TarifaObtenerItem");
                Conexion.AddParemeter("TarifaId", SqlDbType.Int, 10, ParameterDirection.Input, TarifaId);
                SqlDataReader reader = Conexion.ReaderData();
                FillSchemeTable(reader);
                while (reader.Read())
                {
                    TarifaEntity entity = new TarifaEntity();
                    if (FillFrom<TarifaEntity>(reader, entity)) ListaData.Add(entity);
                }
                reader.Close();
                return ListaData;
            }
            catch (Exception ex)
            {
                Conexion.Restore();
                throw ex;
            }
        }

        public List<TarifaEntity> BuscarRecurso(String Nombre,int Tipo)
        {
            try
            {
                List<TarifaEntity> ListaData = new List<TarifaEntity>();
                Conexion.OpenConnection();
                Conexion.Store("sp_TarifaBuscarRecurso");
                Conexion.AddParemeter("Nombre", SqlDbType.VarChar, 50, ParameterDirection.Input, Nombre);
                Conexion.AddParemeter("Tipo", SqlDbType.Int, 10, ParameterDirection.Input, Tipo);
                SqlDataReader reader = Conexion.ReaderData();
                FillSchemeTable(reader);
                while (reader.Read())
                {
                    TarifaEntity entity = new TarifaEntity();
                    if (FillFrom<TarifaEntity>(reader, entity)) ListaData.Add(entity);
                }
                reader.Close();
                return ListaData;
            }
            catch(Exception ex)
            {
                Conexion.Restore();
                throw ex;
            }
        }
    }
}
